### PR TITLE
[Sensiolabs] Add @Template annotation to render() Rector

### LINKED ISSUE #287: Refactor annotations
Hey, is it possible to refactor Template() annotation to render method in Symfony?

Before
```
/**
 * @Template()
 * @Route("/index", name="_index")
 */
public function indexAction()
{
    return [];
}
```
After
```
/**
 * @Route("/index", name="_index")
 */
public function indexAction()
{
    return $this->render('index.html.twig');
}
```

### LINKED ISSUE #295: removeAnnotationFromNode regenerates annotations with unnecessary space
Hey, when I call `$this->docBlockAnalyzer->removeAnnotationFromNode($classMethodNode, 'Template');` template annotation is removed correctly, but it converts `@Route("/", name="homepage")` to `@Route ("/", name="homepage")`. Notice that space created there.

### PR DESCRIPTION
**Pull request by TomasVotruba**
Closes #287 

Closes #295 

### Resources

- https://github.com/sensiolabs/SensioFrameworkExtraBundle/blob/master/Templating/TemplateGuesser.php
- https://github.com/sensiolabs/SensioFrameworkExtraBundle/blob/master/Tests/Templating/TemplateGuesserTest.php

- Docs: https://symfony.com/doc/4.0/bundles/SensioFrameworkExtraBundle/annotations/view.html

### CODE REVIEW

**Comment by TomasVotruba** [comment]
@mssimi Could I ask you for a review?
And also providing more use cases from your real practise?

I think we'll have to reimplement `TemplateGuesser` in framework agnostics way here

**Comment by mssimi** [comment]
I will test it on my code.

**Comment by mssimi** [comment]
I am unable to run rector of this branch. With dev-master I have no problem.

`Class 'Symplify\PackageBuilder\Console\Style\SymfonyStyleFactory' not found in /home/mssimi/www/xxx/vendor-bin/rector/vendor/rector/rector/bin/rector:47`

**Comment by TomasVotruba** [comment]
> I am unable to run rector of this branch. With dev-master I have no problem.

I've just rebased on `master`, there was one autoloading fix

**Comment by TomasVotruba** [comment]
@mssimi I've added framework agnostic `TemplateGuesser` to cover your cases. Check it :+1:

**Comment by mssimi** [comment]
It is no good yet:)

```php
return $this->redirectToRoute('_forgotten'); // before
return $this->render('AppBundle:User:forgotten.html.twig'); // after
```

```php
// before
return $this->render('@App/User/changePassword.html.twig', array(
     'form' => $form->createView()
));
//after
return $this->render('AppBundle:User:passwordChange.html.twig');
```

It should touch only return array statements.

**Comment by TomasVotruba** [comment]
Let's fix it :) What is expected value for those 2 cases?

**Comment by mssimi** [comment]
I'm writing tests for this use cases. Should I create pull request as I did before or is there any way to add commits to this pull request?

**Comment by TomasVotruba** [comment]
>  Should I create pull request as I did before or is there any way to add commits to this pull request?

Yes, that was great for my workflow

**Comment by mssimi** [comment]
I forgot to added these to TemplateAnnotationRectorTest sorry:)

**Comment by TomasVotruba** [comment]
> I forgot to added these to TemplateAnnotationRectorTest sorry:)

No problem, I'll handle that. The wrong/correct files are the most important addition :+1:

**Comment by mssimi** [comment]
Just one off topic question. Can rector rename and move templates? I have several ideas how this could help me, actually I have multiple ideas, I guess I will look more into it:)

**Comment by TomasVotruba** [comment]
@mssimi I though about adding yml and other files - but it's too complicated and require lot of extra work which would influence the quality of PHP part, so [I've removed it](https://github.com/rectorphp/rector/pull/206).

But if you've got some Symfony/PHP related ideas, I'd be happy to help.

>  I guess I will look more into it:)

The best place to start with is ["How it works"](https://github.com/rectorphp/rector/blob/master/docs/HowItWorks.md) in docs.

If you have plenty of ideas and want to discuss which one to pick, reach me out on hangouts on tomas.vot@gmail.com :+1:

**Comment by TomasVotruba** [comment]
Test are running now. Feel free to send more cases in PR

**Comment by mssimi** [comment]
I checked it quickly and still found some problems, but we r getting there:). I will add more commits later today.

**Comment by TomasVotruba** [comment]
@mssimi Could you send PR with failing test cases like before?

**Comment by mssimi** [comment]
@TomasVotruba I tried to re-implement it myself, there is bug in phpdoc regeneration, it generates unneeded space, can't find it. Are you available on hangout to discuss it?

**Comment by TomasVotruba** [comment]
Sure: tomas.vot@gmail.com

File: src/config/level/symfony/symfony33.yml [hunk 1]
```diff
@@ -25,9 +25,6 @@ rectors:
         # debug
         'Symfony\Component\Debug\Exception\ContextErrorException': 'ErrorException'
 
-        # http-kernel
-        Rector\Rector\Contrib\Symfony\HttpKernel\TemplateAnnotationRector: ~
```
Comment (TomasVotruba) [inline src/config/level/symfony/symfony33.yml]: Ha, a bug :)

**Comment by TomasVotruba** [comment]
@mssimi Could you review?

I've added support for `TemplateGuesser` for v5 and v3

**Comment by mssimi** [comment]
@TomasVotruba I already refactored one project with that, everything seems fine:)

**Comment by TomasVotruba** [comment]
I've added `@Route (` space solution as well.

Thanks for fast cooperation :+1:

**Pull request merged**